L1Trigger/L1TMuonCPPF/src/RecHitProcessor.cc

```diff
@@ -171,30 +171,31 @@ void RecHitProcessor::processLook(const edm::Event &iEvent,
         // Condition to save the CPPFDigi
         if (((*cppf1).rawId == rawId) && ((*cppf1).strip == rechitstrip)) {
           int old_strip = (*cppf1).strip;
-          int before = 0;
-          int after = 0;
+          int before = (*cppf1).strip;
+          int after = (*cppf1).strip;
 
-          if (cppf1 != CppfVec1.begin())
+          if (std::distance(CppfVec1.begin(), cppf1) >= 2)
             before = (*(cppf1 - 2)).strip;
-          else if (cppf1 == CppfVec1.begin())
-            before = (*cppf1).strip;
-          if (cppf1 != CppfVec1.end())
+
+          if (std::distance(cppf1, CppfVec1.end()) > 2)
             after = (*(cppf1 + 2)).strip;
-          else if (cppf1 == CppfVec1.end())
-            after = (*cppf1).strip;
+
           cppf = cppf1;
 
           if (clustersize == 2) {
+            bool can_move_back = cppf1 != CppfVec1.begin();
```

Comment:
Out of curiosity should this be the same as: https://github.com/cms-sw/cmssw/blob/e5f6686c8e8aea3342ab98c45f0ad8503d24ffcf/L1Trigger/L1TMuonCPPF/src/RecHitProcessor.cc#L177 ?

Reply:
Kind of - `if (std::distance(CppfVec1.begin(), cppf1) >= 1)`

Reply:
We only move by 1 in both directions, so I think comparing to begin/end is more clear